requirements.dev.txt

```diff
@@ -4,3 +4,7 @@ sphinx-autodoc-typehints
 recommonmark
 pypandoc
 twine
+
+pytest
```

Comment:
it's actually a bit of a misnomer. Currently it's used by the document build server to install dependencies needed for generating the documents. Generally the idea was for it to hold anything that is not a direct project dependency or a test dependency, which currently means doc generation dependencies. Maybe we should rename it to requirements.doc.txt.

Now when running setup.py tests all test dependencies will be installed automatically. This should also include virtualenvs. However, IDEs may or may not parse the setup.py for tests_require.

Anyway, not sure what the best setup is. Maybe a requirements.doc.txt for the documentation build server and a requirements.dev.txt with test dependencies (and maybe twine). Could we leave requirements.dev.txt as is for now and then separately decide whether to split it up into doc/dev?